src/AgentModel.cpp

```diff
@@ -155,206 +159,261 @@ void AgentModel::decisionProcessStop() {
+                        stop = true;
                         continue;
                     }
-
-                    // ego has a yield sign in front
-                    if (e.type == agent_model::SignalType::SIGNAL_YIELD)
+                    
+                    // right target and right turn -> continue
+                    else if (t.position == agent_model::TARGET_ON_RIGHT && _input.vehicle.maneuver == agent_model::Maneuver::TURN_RIGHT)
                     {
-                        // ignore if target has no prority
-                        if (t.priority == agent_model::TargetPriority::TARGET_PRIORITY_NOT_SET 
-                        || t.priority == agent_model::TargetPriority::TARGET_ON_GIVE_WAY_LANE)
-                            continue;
-
-                        // ignore if target is too far away from junction
-                        // double tJunctionTarget = t.dsIntersection / t.v;
-                        // TODO: do something with both time variables...
-                        // TODO: maybe check if target will have passed intersection
-
-                        // Mark as relevant if target is on intersection
-                        if (t.priority == agent_model::TargetPriority::TARGET_ON_INTERSECTION)
-                        {
-                            ii++;
-                            continue;
-                        }
-
-                        //
-                        if (t.priority == agent_model::TargetPriority::TARGET_ON_PRIORITY_LANE 
-                        && t.dsIntersection <= t.v * _param.stop.TMax)
-                        {
-                            ii++;
-                            continue;
-                        }
+                        continue;
                     }
 
-                    // check if driver has to wait when turning left
-                    if (e.type == agent_model::SignalType::SIGNAL_PRIORITY 
-                    && _input.vehicle.maneuver == agent_model::Maneuver::TURN_LEFT)
+                    // if no special case, check if ego reaches junction earlier
+                    if (rel->ds / _input.vehicle.v < t.dsIntersection / t.v) 
                     {
                         continue;
+                    } 
+                    // if target vehicle approaches intersection earlier
+                    else 
+                    {
+                        stop = true;
+                        continue;
                     }
                 }
-                if (ii == 0)
-                    continue;
 
-                // add stop point
-                _state.decisions.stopping[i].id = e.id;
-                _state.decisions.stopping[i].position = _input.vehicle.s + ds;
-                _state.decisions.stopping[i].standingTime = _param.stop.tSign;
+                // stop for target already on intersection
+                if (t.priority == agent_model::TargetPriority::TARGET_ON_INTERSECTION)
```

Comment:
Maybe use TargetPosition here? Then we could delete this entry from the enum and wouldn't have two redudant (and possibly confusing) fields in the interface